### PR TITLE
build test binary with bazel

### PR DESCRIPTION
**Pull request by mikedanese**
I think... this should work

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -1,7 +1,7 @@
 git_repository(
     name = "io_bazel_rules_go",
-    commit = "c7e02c851a8faf803f260800e7387248ccd47683",
-    remote = "https://github.com/bazelbuild/rules_go.git",
+    commit = "dbe525c13b24c8f0984f51351e63c0c3d774db59",
+    remote = "https://github.com/mikedanese/rules_go.git",
```
Comment (spxtr) [inline WORKSPACE]: This is a huge red flag.

Comment (mikedanese) [inline WORKSPACE]: This is temporary pending a change in rules_go.  I have done this on previous PRs that requires joint changes in rules_go and always revert before final merge.

Comment (spxtr) [inline WORKSPACE]: Waiting on https://github.com/bazelbuild/rules_go/pull/229?

Comment (mikedanese) [inline WORKSPACE]: yup

**Comment by ixdy** [comment]
LGTM as soon as you get the upstream go rule merged. (Looks like that one just needs rebase and it's good to go.)

**Comment by spxtr** [comment]
LGTM, please rebase

**Comment by mikedanese** [comment]
@k8s-bot bazel test this.

**Comment by mikedanese** [comment]
@k8s-bot bazel test this

**Comment by mikedanese** [comment]
We need https://github.com/bazelbuild/rules_go/pull/241

File: test/e2e/BUILD [hunk 2]
```diff
@@ -240,6 +237,14 @@ go_library(
     ],
 )
 
+genrule(
```
Comment (mikedanese) [inline test/e2e/BUILD]: I can leave this genrule out if we don't want to preserve the name of the test binary.

File: test/e2e/BUILD [hunk 3]
```diff
@@ -240,6 +237,14 @@ go_library(
...
+    name = "gen_e2e.test",
+    testonly = 1,
+    srcs = [":go_default_test"],
+    outs = ["e2e.tests"],
```
Comment (ixdy) [inline test/e2e/BUILD]: this should be singular (`e2e.test`) to match what `go test -c` produces.

File: test/e2e/BUILD (hunk 2 above)
Comment (ixdy) [inline test/e2e/BUILD]: it'd probably simplify our scripts to keep this.

File: test/e2e/BUILD [hunk 4]
```diff
@@ -218,18 +219,14 @@ go_library(
...
-# TODO(mikedanese): fix @io_bazelbuild/rules_go
-go_library(
+go_test(
     name = "go_default_test",
     srcs = [
         "e2e_test.go",
         "metrics_grabber_test.go",
     ],
     library = ":go_default_library",
     tags = ["automanaged"],
```
Comment (ixdy) [inline test/e2e/BUILD]: probably want to make this manual, so that `go test` doesn't hit it?

Comment (mikedanese) [inline test/e2e/BUILD]: I can add the "integration" tag which we use elsewhere to skip these types of tests. automanagement is still nice for integration tests.

Comment (ixdy) [inline test/e2e/BUILD]: yeah, I meant add a tag to prevent `bazel test ...` from hitting this.

Comment (spxtr) [inline test/e2e/BUILD]: > I can add the "integration" tag which we use elsewhere to skip these types of tests. automanagement is still nice for integration tests.

I think he means adding "manual" but also keeping "automanaged".

**Comment by ixdy** [comment]
/lgtm

I guess this has to be its own line?

**Comment by spxtr** [comment]
All slash commands need to be on their own line.

**Pull request merged**